html/browsers/browsing-the-web/back-forward-cache/resources/helper.sub.js

```diff
@@ -0,0 +1,104 @@
+// Helpers called on the main test HTMLs.
+// Scripts in `send()` arguments are evaluated on the executors
+// (`executor.html`), and helpers available on the executors are defined in
+// `executor.html`.
+
+const idThis = token();
+
+const originSameOrigin =
+  location.protocol === 'http:' ?
+  'http://{{host}}:{{ports[http][0]}}' :
+  'https://{{host}}:{{ports[https][0]}}';
+const originSameSite =
+  location.protocol === 'http:' ?
+  'http://{{host}}:{{ports[http][1]}}' :
+  'https://{{host}}:{{ports[https][1]}}';
+const originCrossSite =
+  location.protocol === 'http:' ?
+  'http://{{hosts[alt][www]}}:{{ports[http][0]}}' :
+  'https://{{hosts[alt][www]}}:{{ports[https][0]}}';
+
+const executorPath =
+  '/html/browsers/browsing-the-web/back-forward-cache/resources/executor.html?uuid=';
+const backPath =
+  '/html/browsers/browsing-the-web/back-forward-cache/resources/back.html';
+
+// On the executor with uuid `idTarget`: Evaluates the script `expr`, and
+// On the caller: returns a Promise resolved with the result of `expr`.
+// This assumes the result can be serialized by JSON.stringify().
+async function evalOn(idTarget, expr) {
+  await send(idTarget, `await send('${idThis}', JSON.stringify(${expr}));`);
```

Comment:
I'm worried by this quoting, there's a lot of quoted code (see also asyncEvalOn). Syntax errors in `$expr` are going to just result in silence here right? I think it would be better if the lowest level API was something like

```js
function call(idTarget, functionToCall, args)
```

where `functionToCall` to call could be 1 of a limited number, `exec`, `eval`, `asyncEval` that way if there's a syntax error in the code, we can send back an error response. Right now if there's a syntax error in the code, I think the `await` will just hang, right?

Reply:
The concern around the error handling seems valid, evalOn() with invalid expression would hang.
I'll fix it and add tests.

Reply:
I changed the interface to `RemoteContext.execute_script(fn, args)` based on RFC 89 which avoids the syntax error issues.